go/libkb/merkle_client.go

```diff
@@ -412,21 +694,69 @@ func (mc *MerkleClient) findValidKIDAndSig(root *MerkleRoot) (keybase1.KID, stri
 	return nilKID, "", MerkleClientError{"no known verifying key"}
 }
 
-func (mc *MerkleClient) verifyRoot(ctx context.Context, root *MerkleRoot, seqnoWhenCalled Seqno) error {
+func (mc *MerkleClient) verifySkipSequence(ctx context.Context, ss SkipSequence, thisRoot *MerkleRoot, lastRoot *MerkleRoot) (err error) {
+	defer mc.G().CTrace(ctx, "MerkleClient#verifySkipSequence", func() error { return err })()
+
+	// In this case, the server did not return a skip sequence. It's OK if
+	// the last known root is too old. It's not OK if the last known root is
+	// from after the server starting providing skip pointers.
+	if ss == nil {
+		mc.G().Log.CDebugf(ctx, "| nil SkipSequence")
+		fss := mc.FirstSeqnoWithSkips()
+		if lastRoot == nil {
+			mc.G().Log.CDebugf(ctx, "| lastRoot==nil, so OK")
+			return nil
+		}
+		if fss < 0 {
+			mc.G().Log.CDebugf(ctx, "| no known root with skips, so OK")
+			return nil
+		}
+		if fss > lastRoot.Seqno() {
+			mc.G().Log.CDebugf(ctx, "| lastRoot (%d) is from before first known root with skips (%d), so OK", int(lastRoot.Seqno()), int(fss))
+			return nil
+		}
+		if thisRoot != nil && lastRoot.Seqno() == thisRoot.Seqno() {
+			mc.G().Log.CDebugf(ctx, "| thisRoot is the same as lastRoot (%d), so OK", int(lastRoot.Seqno()))
+			return nil
+		}
```

Comment:
Is this last check still necessary? We should be returning an empty-but-non-nil list of skips in this case now right?